tests/infrastructure/tekton/conftest.py

```diff
@@ -262,21 +308,93 @@ def pipelinerun_from_pipeline_template(
 
 
 @pytest.fixture()
-def final_status_pipelinerun(pipelinerun_from_pipeline_template):
-    try:
-        for sample in TimeoutSampler(
-            wait_timeout=TIMEOUT_50MIN,
-            sleep=TIMEOUT_1MIN,
-            func=lambda: pipelinerun_from_pipeline_template.instance.status.conditions[0],
-        ):
-            if sample and sample["status"] != Resource.Condition.Status.UNKNOWN:
-                # There are 3 conditions.status possible : Unknown, False, True.
-                LOGGER.info(f"PipelineRun Condition : {sample}")
-                return sample
-
-    except TimeoutExpiredError:
-        LOGGER.error(
-            f"Pipelinerun: {pipelinerun_from_pipeline_template.name} , "
-            f"Preparing for VM teardown due to Timeout Error.Last available sample: {sample}"
-        )
-        raise
+def quay_disk_uploader_secret(custom_pipeline_namespace):
+    with Secret(
+        name="quay-disk-uploader-secret",
+        namespace=custom_pipeline_namespace.name,
+        accesskeyid=base64_encode_str(os.environ["QUAY_ACCESS_KEY_TEKTON_TASKS"]),
+        secretkey=base64_encode_str(os.environ["QUAY_SECRET_KEY_TEKTON_TASKS"]),
+    ) as quay_disk_uploader_secret:
+        yield quay_disk_uploader_secret
+
+
+@pytest.fixture()
+def created_fedora_dv_for_disk_uploader(admin_client, golden_images_namespace, custom_pipeline_namespace):
+    with create_dv(
+        dv_name="fedora-dv-disk-uploader",
+        namespace=custom_pipeline_namespace.name,
+        source=PVC,
+        source_pvc=DataSource(
+            name=OS_FLAVOR_FEDORA, namespace=golden_images_namespace.name
+        ).instance.spec.source.pvc.name,
+        source_namespace=golden_images_namespace.name,
+        size="35Gi",
+        client=admin_client,
+        storage_class=py_config["default_storage_class"],
+    ) as dv:
+        dv.wait_for_dv_success()
+        yield dv
+
+
+@pytest.fixture()
+def vm_for_disk_uploader(custom_pipeline_namespace, admin_client, created_fedora_dv_for_disk_uploader):
+    with VirtualMachineForTests(
+        name="fedora-vm-diskuploader",
+        namespace=custom_pipeline_namespace.name,
+        client=admin_client,
+        data_volume=created_fedora_dv_for_disk_uploader,
+        vm_instance_type=VirtualMachineClusterInstancetype(name="u1.small"),
+        vm_preference=VirtualMachineClusterPreference(name=OS_FLAVOR_FEDORA),
+        run_strategy=VirtualMachine.RunStrategy.ALWAYS,
+    ) as vm:
+        wait_for_running_vm(vm=vm, wait_for_interfaces=False, check_ssh_connectivity=False)
+        yield vm
+
+
+@pytest.fixture()
+def stopped_vm_for_disk_uploader(vm_for_disk_uploader):
+    vm_for_disk_uploader.stop(wait=True)
```

Comment:
Do you need the vm to come to running state and then stop for this test?

Reply:
I want to try a use case where VM is running and pvc in use . https://issues.redhat.com/browse/CNV-54663
Currently we just keep waiting till pipeline timeout. When we have this fix , I should be able to cover that as well in automation. That's the reason i want a running VM.

Reply:
When you add that automation, it would make sense. Currently this code would spend extra time to get the vm in running state and then stopping. We should change this.